client.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"cloud.google.com/go/firestore"
+	"context"
+	"github.com/tslamic/go-oauth2-firestore/token"
+	"time"
+)
+
+type fstore struct {
+	c *firestore.Client
+	n string
+	t time.Duration
+}
+
+func (f *fstore) Put(token *token.Info) error {
+	ctx, cancel := context.WithTimeout(context.Background(), f.t)
+	defer cancel()
+	_, _, err := f.c.Collection(f.n).Add(ctx, token)
+	return err
+}
+
+func (f *fstore) Get(key string, val interface{}) (*token.Info, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), f.t)
+	defer cancel()
+	iter := f.c.Collection(f.n).Where(key, "==", val).Limit(1).Documents(ctx)
```

Comment:
The key we're fetching the document by, is that also the ID? 
If so, perhaps we could just fetch it directly? 
e.g
`dsnap, err := client.Collection("cities").Doc("SF").Get(ctx)`

Then we could also skip the inter.Next() part.

https://firebase.google.com/docs/firestore/query-data/get-data

Reply:
Spot the `Where` - we don't have the document ID, only the key and value we need to search for.

Reply:
👍